test/docker/scylladb.dockerfile

```diff
@@ -1,4 +1,4 @@
-ARG SCYLLA_VERSION=2.3.0
+ARG SCYLLA_VERSION=2.3.1
```

Comment:
@rbino we need to also change this in `.github/workflows/main.yml` before merging.